src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -2241,4 +2241,64 @@ protected static function getAttributeMarkedMutatorMethods($class)
             return false;
         })->map->name->values()->all();
     }
+
+    /**
+     * Check if a given value is cast probably according to the attribute cast type.
+     *
+     * @param  string  $attribute
+     * @return bool
+     */
+    public function isAttributeValueCastedProperly($attribute, $value)
+    {
+        if (! $this->hasCast($attribute)) {
+            return true; // Attribute is not cast, so it's always considered correct
+        }
+
+        $castType = $this->getCastType($attribute);
+
+        switch ($castType) {
```

Comment:
use `match`

Reply:
I didn't notice that this new method still is there. I didn't use it.

Reply:
i removed the `isAttributeValueCastedProperly` method.